spyro/solvers/wave.py

```diff
@@ -315,3 +290,51 @@ def set_last_solve_as_real_shot_record(self):
         if self.current_time == 0.0:
             raise ValueError("No previous solve to set as real shot record.")
         self.real_shot_record = self.forward_solution_receivers
+    
+    @abstractmethod
+    def _set_vstate(self, vstate):
+        pass
+
+    @abstractmethod
+    def _get_vstate(self):
+        pass
+
+    @abstractmethod
+    def _set_prev_vstate(self, vstate):
+        pass
+
+    @abstractmethod
+    def _get_prev_vstate(self):
+        pass
+
+    @abstractmethod
+    def _set_next_vstate(self, vstate):
+        pass
+
+    @abstractmethod
+    def _get_next_vstate(self):
+        pass
```

Comment:
I really like what you have done here. Your changes significantly improve the readability and maintainability of the subsequent time integration methods. However, to ensure there are no unintended increases in memory usage or runtime, I recommend we generate flame graphs for verification.

Reply:
I believe it would not increase the runtime considerably because the numerical parts of the code are probably the bottlenecks. I will work on the flame graphs to make sure that these modifications did not degrade the code performance.

Reply:
It's been a while since I last did a flame graph, but I found this resource from the Firedrake project to be quite useful for code optimization: https://www.firedrakeproject.org/optimising.html. You don't have to do it, you can just time the 3d test and the time_convergence test locally to see if nothing major changed. When we go to elastic 3D, we should generate flamegraphs to try and find any unnecessary bottlenecks